app/modules/submissions/models.py

```diff
@@ -239,6 +239,38 @@ def git_clone(self, project):
 
         return repo
 
+    def import_tus_files(self, transaction_id=None):
+        self.ensure_repository()
+        # @todo duplicates Jons latest tus_upload_dir() to be replaced when available
+        if transaction_id:
+            upload_dir = os.path.join(
+                '_db', 'uploads', '-'.join(['trans', transaction_id])
+            )
+        else:
+            upload_dir = os.path.join('_db', 'uploads', '-'.join(['sub', str(self.guid)]))
+        submission_abspath = self.get_absolute_path()
+        submission_path = os.path.join(submission_abspath, '_submission')
+        num_files = 0
+        for root, dirs, files in os.walk(upload_dir):
+            num_files = len(files)
+            for name in files:
+                log.debug(
+                    'moving upload %r to sub dir %r'
+                    % (
+                        name,
+                        submission_path,
+                    )
+                )
+                os.rename(os.path.join(root, name), os.path.join(submission_path, name))
+
+        if num_files > 0:
+            log.info('Tus collect for %d files moved' % (num_files))
+            self.git_commit('Tus collect commit for %d files.' % (num_files,))
+            self.git_push()
+
+        if transaction_id:
+            os.rmdir(upload_dir)
+
```

Comment:
can we have this return a list of _what_ was imported?  ideally would be *Assets*, but that may be a little tricky since that is done so behind the scenes.  but at minimum a list of the file paths?

on that note, maybe we need a way to filter on a bunch of assets by file paths!  :smile:

Reply:
I need more coffee to be sure but I think this should create the submission too and return that. 
Again, see slack